src/cargo/core/workspace.rs

```diff
@@ -201,6 +201,17 @@ impl<'cfg> Workspace<'cfg> {
         }.parent().unwrap()
     }
 
+    /// Returns the root path of this workspace.
+    ///
+    /// That is, this returns the path of the directory containing the
+    /// `Cargo.toml` which is the root of this workspace.
+    pub fn workspace_dir(&self) -> Option<&Path> {
+        match self.root_manifest {
```

Comment:
Hm so in some sense *all* crates are a member of a workspace regardless of configuration, it's just that some workspaces only contain one member. My feeling is that code wishing to use this would basically othrewise be getting this env var unwarp_or CARGO_MANIFEST_DIR. 

@shepmaster mind weighing in though? Do you have thoughts on what you'd do if a build script
*isn't* in a workspace?